test/e2e-go/features/incentives/challenge_test.go

```diff
@@ -47,10 +46,17 @@ func eligible(address string) bool {
 func TestChallenges(t *testing.T) {
 	partitiontest.PartitionTest(t)
 	defer fixtures.ShutdownSynchronizedTest(t)
-
-	t.Parallel()
 	a := require.New(fixtures.SynchronizedTest(t))
 
+	retry := true
+	for retry {
+		retry = testChallengesOnce(t, a)
+	}
```

Comment:
```suggestion
	retries := 0
	for testChallengesOnce(t, a) && retries < 10 {
		retries += 1
	}
```

Reply:
Wouldn't this change lead to the test passing if it retried too many times?

These tests all run with a timeout from `go test`, so I was reluctant to add extra code to do similar.  Let's just fail for overrunning the timeout in the case of too many retries.